cromwell_tools/cromwell_tools.py

```diff
@@ -1,34 +1,144 @@
+            statuses.append('Unknown')
+        else:
+            response_json = response.json()
+            status = response_json['status']
+            statuses.append(status)
+            timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+            print('{0} workflow {1}: {2}'.format(timestamp, id_, status))
+    return statuses
+
+
+def wait_until_workflow_completes(
+        cromwell_url, workflow_ids, timeout_minutes, poll_interval_seconds=30, cromwell_user=None,
+        cromwell_password=None, secrets_file=None):
+    """
+    given a list of workflow ids, wait until cromwell returns successfully for each status, or
+    one of the workflows fails or is aborted.
+
+    :param list workflow_ids:
+    :param int timeout_minutes:
+    :param int poll_interval_seconds: number of seconds between checks for workflow completion
+      (default 30)
+    :param str cromwell_url:
+    :param str cromwell_user:
+    :param str cromwell_password:
+    :param str secrets_file:
+    :return:
+    """
+    cromwell_user, cromwell_password = harmonize_credentials(
+        secrets_file, cromwell_user, cromwell_password)
+    start = datetime.now()
+    timeout = timedelta(minutes=int(timeout_minutes))
+    while True:
+        if datetime.now() - start > timeout:
+            msg = 'Unfinished workflows after {0} minutes.'
+            raise Exception(msg.format(timeout))
+        statuses = get_workflow_statuses(workflow_ids, cromwell_url, cromwell_user, cromwell_password)
+        all_succeeded = True
+        for i, status in enumerate(statuses):
+            if status in _failed_statuses:
+                raise Exception('Stopping because workflow {0} {1}'.format(workflow_ids[i], status))
+            elif status != 'Succeeded':
+                all_succeeded = False
+        if all_succeeded:
+            print('All workflows succeeded!')
+            break
+        else:
+            time.sleep(poll_interval_seconds)
+
+
+def start_workflow(
+        wdl_file, inputs_file, url, options_file=None, inputs_file2=None, zip_file=None, user=None,
+        password=None):
     """Use HTTP POST to start workflow in Cromwell.
 
     :param _io.BytesIO wdl_file: wdl file.
-    :param _io.BytesIO zip_file: zip file.
     :param _io.BytesIO inputs_file: inputs file.
-    :param _io.BytesIO inputs_file2: inputs file 2.
-    :param _io.BytesIO options_file: configs file.
-    :param ListenerConfig green_config: The ListenerConfig class of current app.
+    :param _io.BytesIO options_file: (optional) cromwell configs file.
+    :param _io.BytesIO inputs_file2: (optional) inputs file 2.
+    :param _io.BytesIO zip_file: (optional) zip file containing dependencies.
+    :param str url: cromwell url
+    :param str user: cromwell username
+    :param str password: cromwell password
+
     :return requests.Response response: HTTP response from cromwell.
     """
     files = {
         'wdlSource': wdl_file,
         'workflowInputs': inputs_file,
```

Comment:
There is a comma left behind, although python works for this.

Reply:
See comment below to dave.